packages/generator/templates/app/app/auth/pages/forgot-password.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { useState } from "react"
+import { BlitzPage, useRouterQuery, Link } from "blitz"
+import Layout from "app/layouts/Layout"
+import { LabeledTextField } from "app/components/LabeledTextField"
+import { Form, FORM_ERROR } from "app/components/Form"
+import {
+  ForgotPasswordInput,
+  ForgotPasswordInputType,
+  ResetPasswordInput,
+  ResetPasswordInputType,
+} from "app/auth/validations"
+import forgotPassword from "app/auth/mutations/forgotPassword"
+import resetPassword from "app/auth/mutations/resetPassword"
+
+/*
+ * By default this page renders the ForgotPasswordForm form which gets the user
+ * email and sends password reset instructions.
+ *
+ * When they click the link in that email, they will be sent back to this same
+ * page but with the token set as a query paramater like `?token=XXXXX`.
+ *
+ * The ResetPasswordForm is rendered if the `?token=` paramater is present in the URL
+ */
+
+const ForgotPasswordForm = () => {
+  const [success, setSuccess] = useState(false)
+  return (
+    <div>
+      <h1>Forgot your password?</h1>
+
+      {success ? (
+        <div>
+          <h2>Request Submitted</h2>
+          <p>
+            If your email is in our system, you will receive instructions to reset your password
+            shortly.
+          </p>
+        </div>
+      ) : (
+        <Form<ForgotPasswordInputType>
+          submitText="Send Reset Password Instructions"
+          schema={ForgotPasswordInput}
+          initialValues={{ email: "" }}
+          onSubmit={async (values) => {
+            try {
+              await forgotPassword(values)
+              setSuccess(true)
+            } catch (error) {
+              return {
+                [FORM_ERROR]: "Sorry, we had an unexpected error. Please try again.",
+              }
+            }
+          }}
+        >
+          <LabeledTextField name="email" label="Email" placeholder="Email" />
+        </Form>
+      )}
+    </div>
+  )
+}
+
+const ResetPasswordForm = () => {
```

Comment:
I think the SSR might be getting confused by this page.  When I actually click the link from the reset email, I get this console error on the Set a New Password page: 

react-dom.development.js?2961:67 Warning: Text content did not match. Server: "Forgot your password?" Client: "Set a New Password"
    at h1
    at div
    at ResetPasswordForm (webpack-internal:///./pages/forgot-password.tsx:151:75)
    at ForgotPasswordPage (webpack-internal:///./pages/forgot-password.tsx:288:75)

Reply:
Fixing this by separating this page into two pages